src/validation-engine.js

```diff
@@ -331,6 +340,13 @@ function nodeLabel(node) {
   }
 
   if (node.termType === 'BlankNode') {
+    const pointer = constraint.shapeNodePointer.out(param)
+    const list = pointer.list()
+    if (list) {
+      const items = Array.from(take(3, list))
+      return items.join(', ') + (items.length === 3 ? ' ...' : '')
```

Comment:
Maybe it would be useful to give even more info than just an ellipsis. If it does not affect performance too much

```suggestion
      const { head, count } = take(3, list)
      const items = Array.from(head)
      return items.join(', ') + (items.length === 3 ? ` (and ${count} more)` : '')
```